src/Umbraco.Web.Common/DependencyInjection/UmbracoBuilder.ImageSharp.cs

```diff
@@ -51,6 +52,8 @@ public static IServiceCollection AddUmbracoImageSharp(this IUmbracoBuilder build
                 .SetCache<PhysicalFileSystemCache>()
                 .SetCacheHash<CacheHash>()
                 .AddProvider<PhysicalFileSystemProvider>()
+                .ClearProcessors()  // ImageSharp includes the processors by default, so we have to clear and re-add to control the order
```

Comment:
That's a very good point, I've changed it to only remove the `ResizeWebProcessor`, it's a bit weird though that it seems like it was added by default, so I'm wondering if we're also adding `FormatWebProcessor` and `BackgroundColorWebProcessor` twice, or maybe ImageSharp handles that for us like we have `AddUnique`

Reply:
I've changed it back to clear processors like we talked about, to make sure the order is correct 😄

Reply:
Looks like [`AddProvider`](https://github.com/SixLabors/ImageSharp.Web/blob/15cb36a034d52a18e364e901e028f8991adad4d5/src/ImageSharp.Web/DependencyInjection/ImageSharpBuilderExtensions.cs#L135) uses [`TryAddEnumerable`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.tryaddenumerable?view=dotnet-plat-ext-5.0&viewFallbackFrom=net-5.0), so it won't add the service(s) of the same type more than once 👍🏻

The ImageSharp sample isn't quite clear on this, as it does the same on the [default setup](https://github.com/SixLabors/ImageSharp.Web/blob/d0559520245dffc484cd7fc77ae35fd20f0ff3b9/samples/ImageSharp.Web.Sample/Startup.cs#L43-L62), but clears them on the [custom one](https://github.com/SixLabors/ImageSharp.Web/blob/d0559520245dffc484cd7fc77ae35fd20f0ff3b9/samples/ImageSharp.Web.Sample/Startup.cs#L107-L139).

I think it's same to clear them first though, so we are sure they are added in the correct order. We only need to make sure this is run before any custom code does (e.g. from composers, but I think that will be the case)...

Reply:
Yeah I'm thinking it's best to clear them first to make sure the order is as we want it, and they should be run before any custom code since they're added when we do `.AddBackOffice`, which is before `.AddComposers()`, so you should be able to add your own processors with no issues.

Reply:
Order of processors doesn't matter (and shouldn't). Processors should have unique, non-collision inducing commands.

Reply:
Just to be sure/clear: the order of the processors added to the options/services doesn't matter, but the order in which the commands are added to the query string does, right?

Reply:
Yes, that’s correct. The API is imperative

Reply:
@JimBobSquarePants The ImageSharp.Web API _does_ use the order of processors, as I get the resized image width/height when the `CropWebProcessor` is added last, but the `crop` query string is the first one in the URL. 

The following code confirms this behavior:
https://github.com/SixLabors/ImageSharp.Web/blob/d0559520245dffc484cd7fc77ae35fd20f0ff3b9/src/ImageSharp.Web/Processors/WebProcessingExtensions.cs#L38-L41.

Another thing to note is that this will run every (DI registered) processor, even when none if the commands matches!

Reply:
That’s a bug. Can you please raise an issue.